### ISSUE TITLE
link should not have role=button unless it 

### ISSUE THREAD

**Issue by DavidMacDonald** [twbs/bootstrap#29610]
Sorry @mdo Mark...Duplicate o #13371 should be reopened

```
<a class="btn btn-primary" href="#" role="button">Link</a>
```

If this is a link then don't add the role="button" because a screen reader user will think it DOES someTHING instead of going someWHERE

Check out the aria spec for role=button
https://www.w3.org/TR/wai-aria-1.1/#button

**Comment by patrickhlauke** [comment]
that issue from 2014 refers to an old version of bootstrap, where in the documentation there was a lot of rubbish.

the original issue #13371 also seems to refer specifically to the "active state" section. here, they're not links per se, but toggles that can be turned on or off. so not actual links, and in that case (all other things being equal, i.e. the keyboard activation etc) `role="button"` is in fact appropriate (if it's not possible to directly use a `<button>`)

the latest equivalent documentation is here https://getbootstrap.com/docs/4.3/components/buttons/ and you'll note that, for the first section on button tags for instance, it contains the following:

> When using button classes on `<a>` elements that are used to trigger in-page functionality (like collapsing content), rather than linking to new pages or sections within the current page, these links should be given a `role="button"` to appropriately convey their purpose to assistive technologies such as screen readers.

**Comment by patrickhlauke** [comment]
though, on third reading, that section is a bit...confused, as it's not quite clear to me how, outside of using it in combination with the toggle plugin, the active state in isolation would be needed, let alone then forced and semantically reinforced by using `aria-pressed` as well...might be worth rethinking in the long run.

**Comment by patrickhlauke** [comment]
btw @DavidMacDonald ... use backticks in github comments around markup otherwise your comment appears all munged...

**Comment by marcoscv-work** [comment]
I think the strange thing is there are present `href=""` and `role="button"` at the same time, for a screen reader is clearly a button but it could follow a link..

**Comment by patrickhlauke** [comment]
the example assumes that there's javascript in place that prevents the link from acting as a link (as is common practice when doing those sorts of faked buttons)

**Comment by mdo** [comment]
@patrickhlauke Thoughts on next steps here?

**Comment by patrickhlauke** [comment]
i think this can be closed, as we're making it clear enough in the docs that even though these controls say "Link", it refers to the underlying element used, but that they're acting as buttons.

and we have this already in the page as well

> When using button classes on `<a>` elements that are used to trigger in-page functionality (like collapsing content), rather than linking to new pages or sections within the current page, these links should be given a `role="button"` to appropriately convey their purpose to assistive technologies such as screen readers.

i don't believe there's a big risk of authors misunderstanding this / not understanding the opposite ... that you shouldn't add `role="button"` unless it acts as a button. not seen much evidence of authors cargo-culting code from here and marking links that act as links up as being `role="button"`
